File: archive/BrownBags/EmployeeVisitor/employee_report_visitor.rb
class EmployeeReportVisitor
  def visitHourly(hourlyEmployee)
    name = hourlyEmployee.getName
    hours = hourlyEmployee.getHours
    @reportLine = "Hourly #{name} worked #{hours} hours."   
  end
  
  def visitSalaried(salariedEmployee)
    name = salariedEmployee.getName
    salary = salariedEmployee.getSalary
    @reportLine = "Salaried #{name} earns $#{salary}."
  end
  
  def getReportLine 
    @reportLine
  end
end

File: archive/BrownBags/JavaVisitor/src/EmployeeReportVisitor.java
public class EmployeeReportVisitor implements EmployeeVisitor {
  private String reportLine;

  public String getReportLine() {
    return reportLine;
  }

  public void visit(HourlyEmployee hourlyEmployee) {
    int hours = hourlyEmployee.getHours();
    String name = hourlyEmployee.getName();
    reportLine = String.format("Hourly %s worked %d hours.", name, hours);
  }

  public void visit(SalariedEmployee salariedEmployee) {
    String name = salariedEmployee.getName();
    int salary = salariedEmployee.getSalary();
    reportLine = String.format("Salaried %s earns $%d.", name, salary);
  }
}
